scripts/generate/cli-docs.sh

```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/env bash
+# exit immediately when a command fails
+set -e
+# only exit with zero if all commands of the pipeline exit successfully
+set -o pipefail
+
+if [ "$1" == "operator" ]; then
+  # Print the output via usage but remove the line containing the executable info
+  go run cmd/operator/main.go --help 2> >(grep -v 'Usage of')
+fi
```

Comment:
It is unclear to me where this script is used in the docs generation pipeline (in Makefile). Could you clarify why we need this bash script?

Reply:
Yes, I should have marked this WIP really, sorry for the confusion. 

So this is being called [here](https://github.com/prometheus-operator/prometheus-operator/pull/4446/files#diff-138201325bbc7cecc4d59e865c71badd1692f1dcaa52e02f3e0d596884207321R7).

The reason I have created a script for it is because the `--help` command produces an extraneous line about the executable, which I believe will be different for macOS and linux for example. I was trying to strip that here as it did not appear to work correctly when embedded directly in the markdown.

Furthermore, this script seems to prune that line correctly now in OS X but still the check fails on ubuntu which I need to investigate why that is still producing a diff.

Reply:
I had to rethink this because occasionally, the build info was getting spewed into the markdown since it had to download some modules depending on timing. The latest commit requires us building the binary but avoids us having to create a script since the output is now predictable.